pkg/ocm/authorization.go

```diff
@@ -51,46 +50,39 @@ func (a authorization) AccessReview(ctx context.Context, username, action, resou
 }
 
 func (a authorization) CapabilityReview(ctx context.Context, username, capabilityName, capabilityType string) (allowed bool, err error) {
-	// CapabilityReview is not available yet in ocm-sdk-go:
-	// https://github.com/openshift-online/ocm-sdk-go/blob/master/authorizations/v1/root_client.go
-	// Sending a simple POST request for now.
 	connection, err := a.client.NewConnection()
 	if err != nil {
 		return false, err
 	}
 	defer connection.Close()
 
-	request := connection.Post()
-	request.Path("/api/authorizations/v1/capability_review")
+	capabilityReview := connection.Authorizations().V1().CapabilityReview()
 
-	type CapabilityRequest struct {
-		Name     string `json:"capability"`
-		Type     string `json:"type"`
-		Username string `json:"account_username"`
-	}
-
-	capabilityRequest := CapabilityRequest{
-		Name:     capabilityName,
-		Type:     capabilityType,
-		Username: username,
-	}
-
-	var jsonData []byte
-	jsonData, err = json.Marshal(capabilityRequest)
+	request, err := azv1.NewCapabilityReviewRequest().
+		AccountUsername(username).
+		Capability(capabilityName).
+		Type(capabilityType).
+		Build()
 	if err != nil {
 		return false, err
 	}
-	request.Bytes(jsonData)
 
-	postResp, err := request.SendContext(ctx)
+	postResp, err := capabilityReview.Post().
+		Request(request).
+		SendContext(ctx)
 	if err != nil {
 		return false, err
 	}
 
```

Comment:
See @rollandf  review on OCM SDK. He claims that response.status() should be checked

Reply:
I'm not sure it's required, according to the examples it seems unneeded:
https://github.com/openshift-online/ocm-sdk-go/blob/356e35e01902f8e68e6b2353ffd59ed06becc03a/examples/client_credentials_grant.go#L64

We should get an error for 'status >= 400' on sendConext:
https://github.com/openshift-online/ocm-sdk-go/blob/b3f1fc3136d749c108f319f8af5b1647923f4d49/authorizations/v1/capability_review_client.go#L134

Reply:
@danielerez is right no need for it